api/webhook.go

```diff
@@ -275,64 +201,45 @@ func deleteOutgoingHook(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func regenOutgoingHookToken(c *Context, w http.ResponseWriter, r *http.Request) {
-	if !utils.Cfg.ServiceSettings.EnableOutgoingWebhooks {
-		c.Err = model.NewLocAppError("regenOutgoingHookToken", "api.webhook.regen_outgoing_token.disabled.app_error", nil, "")
-		c.Err.StatusCode = http.StatusNotImplemented
+	props := model.MapFromJson(r.Body)
+
+	id := props["id"]
+	if len(id) == 0 {
+		c.SetInvalidParam("regenOutgoingHookToken", "id")
 		return
 	}
 
-	if !app.SessionHasPermissionToTeam(c.Session, c.TeamId, model.PERMISSION_MANAGE_WEBHOOKS) {
-		c.Err = model.NewLocAppError("regenOutgoingHookToken", "api.command.admin_only.app_error", nil, "")
-		c.Err.StatusCode = http.StatusForbidden
+	hook, err := app.GetOutgoingWebhook(id)
+	if err != nil {
+		c.Err = err
 		return
 	}
 
 	c.LogAudit("attempt")
 
-	props := model.MapFromJson(r.Body)
-
-	id := props["id"]
-	if len(id) == 0 {
-		c.SetInvalidParam("regenOutgoingHookToken", "id")
+	if !app.SessionHasPermissionToTeam(c.Session, hook.TeamId, model.PERMISSION_MANAGE_WEBHOOKS) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_WEBHOOKS)
 		return
 	}
 
-	var hook *model.OutgoingWebhook
-	if result := <-app.Srv.Store.Webhook().GetOutgoing(id); result.Err != nil {
-		c.Err = result.Err
+	if c.TeamId != hook.TeamId && c.Session.UserId != hook.CreatorId && !app.SessionHasPermissionToTeam(c.Session, c.TeamId, model.PERMISSION_MANAGE_OTHERS_WEBHOOKS) {
```

Comment:
I'm probably missing something, given that this line is essentially unchanged from before this PR, but why are we ANDing all three of these checks together? Is it possible that the `TeamId`'s could be the same but the `userId`/`creatorId` different? In that case, should the user be allowed to perform the action regardless of whether they have the `MANAGE_OTHERS_WEBHOOKS` permission?